common/service/host/kcs.c

```diff
@@ -55,14 +60,90 @@ void reset_kcs_ok()
 	proc_kcs_ok = false;
 }
 
+#ifdef ENABLE_PLDM
+
+enum cmd_app_get_sys_info_params {
+	LENGTH_INDEX = 0x05, // skip netfun, cmd code, paramter selctor, set selctor, encoding
+	VERIONS_START_INDEX = 0x06,
+};
+
+static uint8_t init_text_string_value(const char *new_text_string, char *text_strings)
+{
+	const uint8_t NULL_BYTE = 1, EMPTY_TEXT_STRINGS_SIZE = 2;
+	const uint8_t original_size = pldm_smbios_get_text_strings_size(text_strings);
+	const uint8_t new_text_string_len = strlen(new_text_string);
+	uint8_t new_text_string_start_idx = original_size - 1;
+
+	if (original_size == EMPTY_TEXT_STRINGS_SIZE) {
+		new_text_string_start_idx = 0;
+	}
+
+	text_strings = realloc(text_strings, original_size + new_text_string_len + NULL_BYTE);
+	memcpy(text_strings + new_text_string_start_idx, new_text_string, new_text_string_len);
+
+	*(text_strings + new_text_string_start_idx + new_text_string_len) = '\0';
+	*(text_strings + new_text_string_start_idx + new_text_string_len + 1) = '\0';
+	return pldm_smbios_get_text_strings_count(text_strings);
+}
+
+static void init_bios_information(smbios_bios_information *bios_info, char *bios_version)
+{
+	const uint8_t NOT_IMPLEMENTED = 0;
+
+	bios_info->header.type = SMBIOS_BIOS_INFORMATION;
+	bios_info->text_strings = malloc(sizeof(char) * 2);
+	bios_info->text_strings[0] = bios_info->text_strings[1] = '\0';
+
+	/* DSP0134: 12h + number of BIOS Characteristics Extension Bytes.*/
+	bios_info->header.length = 0x12;
+	bios_info->header.handle = 0x0000;
+	bios_info->vendor = init_text_string_value("N/A", bios_info->text_strings);
+	bios_info->bios_version = init_text_string_value(bios_version, bios_info->text_strings);
+	bios_info->bios_starting_address_segment = NOT_IMPLEMENTED;
+	bios_info->bios_release_date = init_text_string_value("N/A", bios_info->text_strings);
+	bios_info->bios_rom_size = NOT_IMPLEMENTED;
+	bios_info->bios_characteristics = NOT_IMPLEMENTED;
+	bios_info->bios_characteristics_extension_bytes = NOT_IMPLEMENTED;
+	bios_info->system_bios_major_release = NOT_IMPLEMENTED;
+	bios_info->system_bios_minor_release = NOT_IMPLEMENTED;
+	bios_info->embedded_controller_firmware_major_release = NOT_IMPLEMENTED;
+	bios_info->embedded_controller_firmware_minor_release = NOT_IMPLEMENTED;
+	bios_info->extended_bios_rom_size = NOT_IMPLEMENTED;
+}
+
+static int update_bios_information(char *bios_version_start_ptr, uint8_t bios_version_len)
+{
+	smbios_bios_information *bios_info = malloc(sizeof(smbios_bios_information));
+
+	char *bios_version_str = malloc(sizeof(char) * bios_version_len + 1);
+	memcpy(bios_version_str, bios_version_start_ptr, bios_version_len);
+	*(bios_version_str + bios_version_len) = '\0';
+
+	init_bios_information(bios_info, bios_version_str);
+
+	int rc = pldm_smbios_set_bios_information(bios_info);
+	if (rc < 0) {
+		return rc;
```

Comment:
This error path leaks significant memory.

Reply:
Done, didn't notice that, thanks for reminding.